OneCommunityGlobal/HighestGoodNetworkApp
# Saideepak equipment details

# Description
Implemented the equipment details page in the bmdashboard as part of the phase 2 development.
![319460727-d1c220e2-a82d-4f65-91ba-9e1c555db8de](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/114311909/c1cc7361-e928-4407-b431-d1b81d9ae703)


## Related PRS (if any):
This frontend PR is related to the [837](https://github.com/OneCommunityGlobal/HGNRest/pull/837) backend PR. To test this frontend PR you have to checkout the branch mentioned in the [837](https://github.com/OneCommunityGlobal/HGNRest/pull/837) backend PR
…

## Main changes explained:
- Created EquipmenrDetails.jsx and EquipmentDetail.css
- Added routing for equipment details
-  Added the url and added pages like equipmentactions, equipmentreducers, equipmentconstants etc
…

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally(npm start)
3. Clear site data/cache
4. log as any user eg: volunteer, owner or admin
5. go to http://localhost:3000/bmdashboard and login again
6. Navigate to http://localhost:3000/bmdashboard/equipment/65c16ef325834114ac9427e0
7. Verify the design from the figma template https://www.figma.com/file/kpwXZP2pLtOJ1Fn2u7Sivn/HGN-PII?type=design&node-id=430-5449&mode=design , you can find the template under the name E/T Detail Page
8. Verify the details of the page from the database from buildingInventoryItems collection.
9. Feel free to change the id in the url to any other id in the database and verify the webpage data.


## Screenshots or videos of changes:
1. Login to BMdashboard
![Screenshot (131)](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/114311909/70c109ef-f4c1-4123-a6f3-0d01dadf0a40)
2. Navigate to equipment detail page
![Screenshot (132)](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/114311909/930afc85-f0f6-4d4b-ad48-3c4e3dfed3e7)


## Note:
The scope of this PR is only limited to a few data fields namely Belongs to Project, Class, Name, Number, Ownership, Rental Duration, Current Usage. All other fields are just populated with hardcoded dummy data as there are no such fields in the database. We are still working on updating the database to appropriate values.
